File: src/main/java/com/itsci/mju/maebanjumpen/model/Hire.java
package com.itsci.mju.maebanjumpen.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.EqualsAndHashCode.Exclude;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.List;

@Entity
@Table(name = "hire")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties(ignoreUnknown = true)
public class Hire {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer hireId;

    @Column(nullable = false)
    private String hireName;

    @Column(nullable = false)
    private String hireDetail;

    @Column(nullable = false)
    private Double paymentAmount;

    @Column(nullable = false)
    private LocalDateTime hireDate;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column(nullable = false)
    private LocalTime startTime;

    @Column(nullable = true)
    private LocalTime endTime;

    @Column(nullable = false)
    private String location;

    @Column(nullable = false)
    private String jobStatus;

    @ElementCollection
    private List<String> progressionImageUrls;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "hirer_id")
    @ToString.Exclude
    @Exclude
    private Hirer hirer;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "housekeeper_id")
    @ToString.Exclude
    @Exclude
    private Housekeeper housekeeper;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "skill_type_id")
    @ToString.Exclude
    @Exclude
    private SkillType skillType;

    @OneToOne(mappedBy = "hire", fetch = FetchType.LAZY)
    @ToString.Exclude
    @Exclude
    private Review review;

//	@ElementCollection
//	@CollectionTable(name = "hire_additional_skills", joinColumns = @JoinColumn(name = "hire_id"))
//	@Column(name = "additional_skill_type_id")
//	private List<Integer> additionalSkillTypeIds;
//
//	public void setReview(Review review) {
//		if (this.review != null && this.review.getHire() != null) {
//			this.review.setHire(null);
//		}
//		this.review = review;
//		if (review != null) {
//			review.setHire(this);
//		}
//	}
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Hire.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime

@Entity
@Table(name = "hire")
@JsonIgnoreProperties(ignoreUnknown = true)
data class Hire(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = 0,

    @Column(nullable = false)
    var hireName: String = "",

    @Column(nullable = false)
    var hireDetail: String = "",

    @Column(nullable = false)
    var paymentAmount: Double = 0.0,

    @Column(nullable = false)
    var hireDate: LocalDateTime? = null,

    @Column(nullable = false)
    var startDate: LocalDate? = null,

    @Column(nullable = false)
    var startTime: LocalTime? = null,

    @Column(nullable = true)
    var endTime: LocalTime? = null,

    @Column(nullable = false)
    var location: String = "",

    @Column(nullable = false)
    var jobStatus: String = "",

    @ElementCollection
    var progressionImageUrls: MutableList<String> = mutableListOf(),

    @Column(name = "hirer_id", insertable = false, updatable = false)
    var hirerId: Long? = null,

    @Column(name = "housekeeper_id", insertable = false, updatable = false)
    var housekeeperId: Long? = null,

    @Column(name = "skill_type_id", insertable = false, updatable = false)
    var skillTypeId: Long? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "hirer_id")
    var hirer: Hirer? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "housekeeper_id")
    var housekeeper: Housekeeper? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "skill_type_id")
    var skillType: SkillType? = null,

    @OneToOne(mappedBy = "hire", fetch = FetchType.LAZY)
    var review: Review? = null
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Hire) return false
        return id != null && id == other.id


    }


    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String = "Hire(hireId=$id, hireName='$hireName', jobStatus='$jobStatus')"
}

